sdk/src/evaluations/eas.ts

```diff
@@ -0,0 +1,70 @@
+import { Offchain, SchemaEncoder } from "@ethereum-attestation-service/eas-sdk";
+import { TypedDataSigner } from "@ethersproject/abstract-signer";
+import { ethers } from "ethers";
+
+import { DEFAULT_CHAIN_ID } from "../constants.js";
+import { InvalidOrMissingError } from "../types/errors.js";
+import { DuplicateEvaluation, EvaluationData } from "../types/evaluation.js";
+
+type EasEvaluatorConfig = {
+  address: string;
+  chainId: number;
+  signer: TypedDataSigner;
+};
+
+export default class EasEvaluator {
+  offChain: Offchain;
+  schemaEncoder: SchemaEncoder;
+  signer: TypedDataSigner;
+
+  constructor({
+    config = { chainId: DEFAULT_CHAIN_ID, address: "", signer: new ethers.VoidSigner("") },
+  }: {
+    config?: EasEvaluatorConfig;
+  }) {
+    this.offChain = new Offchain({ address: config.address, chainId: config.chainId, version: "0.26" });
+    this.schemaEncoder = new SchemaEncoder("uint256 chainId, address contract, uint256 claimId, string uri");
+    this.signer = config.signer;
+  }
+
+  signOfflineEvaluation = async (evaluation: EvaluationData) => {
+    let encodedData;
+    if (evaluation.type === "duplicate") {
+      const data = evaluation.evaluation as DuplicateEvaluation;
```

Comment:
If you use an algebraic data type, you shouldn't need to cast
https://www.npmjs.com/package/ts-adt

Reply:
Nice, wanted to play with that at one point :D

Reply:
Skipped, see https://github.com/hypercerts-org/hypercerts/pull/659#discussion_r1208032695

(implemented custom type assertion)